sdks/go/pkg/beam/runners/dataflow/dataflowlib/execute.go

```diff
@@ -122,3 +135,21 @@ func PrintJob(ctx context.Context, job *df.Job) {
 	}
 	log.Info(ctx, string(str))
 }
+
+type dataflowPipelineResult struct {
+	JobID   string
```

Comment:
Note that an exported field on an unexported struct is still nominally inaccessible. Consider unexporting it, and having a JobID() method instead.

Reply:
Let's have a JobID() method. I think accessing a job ID might be useful in some cases.

Reply:
I had to add the `JobID()` to the `PipelineResult` interface also. But I think it's fine, since most runners have some kind of a job ID.

Reply:
SGTM for JobID. Remember that in Go it's not necessary to return the exact interface that's supported every time. If the dataflow results or the flink results need different method sets, they can export different methods, and just document what are available on a custom results interface, that can be type asserted to.